apps/token/src/routes/staking/node/staking-form.tsx

```diff
@@ -196,43 +197,35 @@ export const StakingForm = ({
     return () => clearInterval(interval);
   }, [formState, client, pubKey, nodeId]);
 
-  if (formState === FormState.Failure) {
-    return <StakeFailure nodeName={nodeName} />;
-  } else if (formState === FormState.Requested) {
-    return (
-      <Dialog
-        title="Confirm transaction in wallet"
-        intent={Intent.Warning}
-        open={true}
-      >
-        <p>{t('stakingConfirm')}</p>
-      </Dialog>
-    );
-  } else if (formState === FormState.Pending) {
-    return <StakePending action={action} amount={amount} nodeName={nodeName} />;
-  } else if (formState === FormState.Success) {
-    return (
-      <StakeSuccess
-        action={action}
-        amount={amount}
-        nodeName={nodeName}
-        removeType={removeType}
-      />
-    );
-  } else if (
-    availableStakeToAdd.isEqualTo(0) &&
-    availableStakeToRemove.isEqualTo(0)
-  ) {
-    if (appState.lien.isGreaterThan(0)) {
-      return <span className="text-red">{t('stakeNodeWrongVegaKey')}</span>;
-    } else {
-      return <span className="text-red">{t('stakeNodeNone')}</span>;
+  const toggleDialog = useCallback(() => {
+    setIsDialogVisible(!isDialogVisible);
+  }, [isDialogVisible]);
+
+  useEffect(() => {
```

Comment:
suggestion (non blocking): its not a good pattern to use useEffect to set state that simply relies on other state, it makes the code harder to follow. In this case can you not just call `setIsDialogVisible` at the start of the `onSubmit` function?

Reply:
great shout, thanks